seacrowd/sea_datasets/burapha_th/burapha_th.py

```diff
@@ -0,0 +1,166 @@
+            features = schemas.image_text_features(label_names=self.label_chr_dig if task == "character" or task == "digit" else self.label_syl)
+        else:
+            raise NotImplementedError()
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        task = self.config.subset_id.split("_")[2]
+
+        _local_path = dl_manager.download_and_extract(_URLS[task]) 
+        train_path, test_path = _local_path["train"], _local_path["test"]
+        if task != "syllable":
+            train_path = os.path.join(train_path, "train")
+            test_path = os.path.join(test_path, "test")
+        else:
+            train_path = os.path.join(train_path, "train-ori")
+            test_path = os.path.join(test_path, "test-ori")
+
+        data_pair = {}
+
+        for dir_name in os.listdir(train_path):
+            dir_name_split = dir_name.split("-")
+            file_names = []
+
+            for file_name in os.listdir(os.path.join(train_path, dir_name)):
+                file_names.append(os.path.join(train_path, dir_name, file_name))
+
+            label = dir_name_split[0]
+            data_pair[label] = file_names
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": data_pair,
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "filepath": data_pair,
+                    "split": "test",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        task = self.config.subset_id.split("_")[2]
+        counter = 0
+
+        for key, imgs in filepath.items():
+            for img in imgs:
+                if self.config.schema == "source":
+                    yield counter, {"id": str(counter), "image_paths": img, "label": [self.label_chr_dig.index(key) if task == "character" or task == "digit" else self.label_syl.index(key)]}
+                elif self.config.schema == "seacrowd_imtext":
+                    yield counter, {
+                        "id": str(counter),
+                        "image_paths": [img],
+                        "texts": None,
+                        "metadata": {
+                            "context": None,
+                            "labels": [self.label_chr_dig.index(key) if task == "character" or task == "digit" else self.label_syl.index(key)],
```

Comment:
also to improve readability by in-list value checking rather than making `or` statement
```suggestion
                            "labels": [self.label_chr_dig.index(key) if task in ["character", "digit"] else self.label_syl.index(key)],
```

Reply:
This looks nicer. Thank you.